src/main/java/games/strategy/triplea/ui/ResourceBar.java

```diff
@@ -0,0 +1,87 @@
+package games.strategy.triplea.ui;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.BorderFactory;
+import javax.swing.JLabel;
+import javax.swing.border.EtchedBorder;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Resource;
+import games.strategy.engine.data.events.GameDataChangeListener;
+import games.strategy.engine.stats.IStat;
+import games.strategy.triplea.Constants;
+
+/**
+ * Panel used to display the current players resources.
+ */
+public class ResourceBar extends AbstractStatPanel implements GameDataChangeListener {
+  private static final long serialVersionUID = -7713792841831042952L;
+
+  private final UiContext uiContext;
+  private IStat[] statsResource;
```

Comment:
Just curious why you chose to use an array here instead of `List<IStat>`?  It seems you're creating a list anyway; it's not clear to me why you're converting it back to an array.

Reply:
No real reason. The class I based this one used an array but looks like a list should be fine.